src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -34,13 +34,13 @@ namespace Microsoft.IdentityModel.Tokens
     /// Creates <see cref="SignatureProvider"/>s by specifying a <see cref="SecurityKey"/> and algorithm.
     /// <para>Supports both <see cref="AsymmetricSecurityKey"/> and <see cref="SymmetricSecurityKey"/>.</para>
     /// </summary>
-    public class SignatureProviderFactory
+    public class CryptoProviderFactory
```

Comment:
That would be yet another thing to have. Currently, SecurityKey class only has CryptoProviderFactory that it can use to create a signature provider or encryption provider. Also both of these providers share the basic attributes - key and algorithm, so it makes sense to have a common wrapper for them.